src/sr_entities_handler.erl

```diff
@@ -79,7 +91,19 @@ content_types_accepted(Req, State) ->
 -spec content_types_provided(cowboy_req:req(), state()) ->
   {[{binary(), atom()}], cowboy_req:req(), state()}.
 content_types_provided(Req, State) ->
-  {[{<<"application/json">>, handle_get}], Req, State}.
+    #{opts := #{path := Path}} = State,
+    {Method, Req2} = cowboy_req:method(Req),
+    try
+        #{metadata := Metadata} = trails:retrieve(Path),
+        AtomMethod = method_to_atom(Method),
+        #{AtomMethod := #{produces := Produces}} = Metadata,
+        Handler = compose_handler_name(AtomMethod),
+        RetList = [{list_to_binary(X), Handler} || X <- Produces],
```

Comment:
Please replace `list_to_binary` by `iolist_to_binary` just in case someone uses binaries for `consumes`.